### PR TITLE
[CARBONDATA-497][SPARK2]fix datatype issue of CarbonLateDecoderRule

### PR DESCRIPTION
**Pull request by QiangCai**
1.  Fix the data type of dictionary dimension to resolve the logical plan 

2. Perfect  translateFilter method to push down more filters to CarbonScanRDD.

3. Add decimal type field to CarbonExample

### CODE REVIEW

**Comment by QiangCai** [comment]
OK to test

**Comment by QiangCai** [comment]
test this please

**Comment by jackylk** [comment]
test this please

**Comment by CarbonDataQA** [comment]
Build Success, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/38/

**Comment by CarbonDataQA** [comment]
Build Success, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/39/

**Comment by chenliang613** [comment]
add to whitelist

**Comment by CarbonDataQA** [comment]
Build Success, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/44/

File: integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonLateDecodeRule.scala [hunk 1]
```diff
@@ -456,76 +487,143 @@ class CarbonLateDecodeRule extends Rule[LogicalPlan] with PredicateHelper {
...
+        if (p.child.isInstanceOf[CarbonDictionaryTempDecoder]) {
+          val tempDecoder = p.child.asInstanceOf[CarbonDictionaryTempDecoder]
+          tempDecoder.attrList.asScala.foreach{attr => tmpAttrMap.put(attr, attr.attr)}
+        }
         val prExps = p.projectList.map { prExp =>
-          prExp.transform {
-            case attr: AttributeReference =>
-              updateDataType(attr, attrMap, allAttrsNotDecode, aliasMap)
-          }
+              prExp.transform {
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonLateDecodeRule.scala]: feels like incorrect indentation

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonExample.scala [hunk 2]
```diff
@@ -105,13 +106,14 @@ object CarbonExample {
       s"""
          | INSERT INTO TABLE carbon_table
          | SELECT shortField, intField, bigintField, doubleField, stringField,
-         | from_unixtime(unix_timestamp(timestampField,'yyyy/M/dd')) timestampField
```
Comment (jackylk) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonExample.scala]: Can you move some of these SQL in the test case

**Comment by jackylk** [comment]
ok to test

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/58/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/59/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/60/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**